### PR TITLE
POST call for rendering Preview html

### PR DESCRIPTION
**Pull request by sharvaniharan**
POST call for posting wikitext and rendering Preview html

**Phab**: https://phabricator.wikimedia.org/T230198

### CODE REVIEW

File: app/src/main/java/org/wikipedia/edit/preview/EditPreviewHtml.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package org.wikipedia.edit.preview;
+
+import androidx.annotation.Nullable;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class EditPreviewHtml {
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/edit/preview/EditPreviewHtml.java]: Is this used?

File: app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.java [hunk 2]
```diff
@@ -167,19 +182,48 @@ with the original Locale (from above)
...
+     * @param title The PageTitle associated with the text being modified.
+     * @param wikiText The text of the section to be shown in the Preview.
+     */
+    public void showPreview(final PageTitle title, final String wikiText) {
+        hideSoftKeyboard(requireActivity());
+        parentActivity.showProgressBar(true);
+        String url = model.getTitle().getWikiSite().uri() + PAGE_HTML_PREVIEW_ENDPOINT + title.getPrefixedText();
+        String postData;
+        try {
+            postData = "wikitext=" + URLEncoder.encode(wikiText, "UTF-8");
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.java]: We have a utility method called `UriUtil.encodeUrl()` that automatically takes care of catching unsupported encoding exceptions, which will never happen.

File: app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.java [hunk 3]
```diff
@@ -65,16 +67,30 @@
...
             }
+
+            @Override public void onPageFinished(WebView view, String url) {
+                super.onPageFinished(view, url);
+                parentActivity.showProgressBar(false);
+                parentActivity.supportInvalidateOptionsMenu();
+                //Save the html received from the the wikitext to mobile-html transform, to use in the savedInstanceState
+                webview.evaluateJavascript(
+                        "(function() { return ('<html>'+document.getElementsByTagName('html')[0].innerHTML+'</html>'); })();",
+                        html -> {
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.java]: How about `document.documentElement.outerHTML`?

**Comment by sharvaniharan** [comment]
Done

**Comment by cooltey** [comment]
@sharvaniharan I am not able to see the edit pencil icon...
and looks like the page is broken.
Screenshot_1574727898

**Comment by sharvaniharan** [comment]
should be fixed now @cooltey. As a few articles might not have still been purged.. here are some you can try: tulip, dahlia, jasmine

**Comment by sharvaniharan** [comment]
@cooltey That is the response previewHtml form the call to the endpoint. I am thinking it won't hurt to show the title there.. however if we feel otherwise we can let @joewalsh know and change that. Either way, this PR makes no changes to affect that.

File: app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.java [hunk 4]
```diff
@@ -170,19 +183,42 @@ with the original Locale (from above)
...
+     * Fetches preview html from the modified wikitext text, and shows (fades in) the Preview fragment,
+     * which includes edit summary tags. When the fade-in completes, the state of the
+     * actionbar button(s) is updated, and the preview is shown.
+     * @param title The PageTitle associated with the text being modified.
+     * @param wikiText The text of the section to be shown in the Preview.
+     */
+    public void showPreview(final PageTitle title, final String wikiText) {
+        hideSoftKeyboard(requireActivity());
+        parentActivity.showProgressBar(true);
+        String url = model.getTitle().getWikiSite().uri() + PAGE_HTML_PREVIEW_ENDPOINT + title.getPrefixedText();
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.java]: One minor thing:

Can we change the `getPrefixedText()` to `getConvertedText()`?

Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.java]: Done.

**Pull request merged**